### PR TITLE
fixed all requests

### PR DESCRIPTION
**Pull request by maharajamihir**
**Shortly describe your proposed changes:**
Tests for BinomialHeap.java

**Link any issues if they have led to changes in the request:**

sorry for creating another pull request... one year into computer science and i still dont get git :/

**Note the (sub-)exercises the tests are meant for:**

* GAD 8 - Binomial Heap: BinomialHeap.java

### CODE REVIEW

**Comment by CrsiX** [comment]
Well, the knowledge how to use `git` and related tools will come over time, for sure. Thanks for addressing the previously mentioned changes :)
I will have a look into it later on to merge this request, if nobody else has something to comment on it, @N0W0RK?

File: GAD 8 - Binomial Heap/tests/binomilia/TreeCounterResult.java
Comment (N0W0RK) [inline GAD 8 - Binomial Heap/tests/binomilia/TreeCounterResult.java:11]: This should be a primitive int.

File: GAD 8 - Binomial Heap/tests/binomilia/TreeCounterResult.java
Comment (N0W0RK) [inline GAD 8 - Binomial Heap/tests/binomilia/TreeCounterResult.java:51]: This should be an addition.

**Review by N0W0RK — CHANGES_REQUESTED** [review]
I like the improvements. I would ask you to fix the remaining codestyle issues and issues I raised.
The handling of git can be learned, if you need help message me on zulip @Konrad Gößmann.

File: GAD 8 - Binomial Heap/tests/binomilia/BinomialHeapTester.java
Comment (maharajamihir) [reply GAD 8 - Binomial Heap/tests/binomilia/BinomialHeapTester.java:83]: Because then there would be only one element in the heap, which would get removed per iteration. The errors (at least in my case) occured due to big heaps and bigger numbers.

Comment (N0W0RK) [reply GAD 8 - Binomial Heap/tests/binomilia/BinomialHeapTester.java:83]: So then why are those loops not connected

**Comment by maharajamihir** [comment]
The codestyle ist complaining about tabs and white characters. I autoformatted it with eclipse. Thats all i can do. I hope this is enough for those simple tests.

File: GAD 8 - Binomial Heap/tests/binomilia/BinomialHeapTester.java
Comment (N0W0RK) [inline GAD 8 - Binomial Heap/tests/binomilia/BinomialHeapTester.java:52]: This will not work when you start the loop at 0. I don't have an issue with negative loop starts if you actually use those values. Right now the min element in the heap is 0

**Review by N0W0RK — CHANGES_REQUESTED** [review]
So most of it is alright. I just have issues with the two open conversations. If you tell me why your code in line #83 works the way you want it, I would also resolve that conversation.
The codestyle issues are with your javadoc, where you have `* ` at the end of the line, where just `*` would be appropriate. The issue with tabs stems from your indentation with spaces in the javadoc, but tabs in code.

**Comment by maharajamihir** [comment]
# Line 83:

The method ```testBigNumbers``` in general just adds a lot of numbers to the binomial heap. It just tests its min on the way. The real magic is when it starts deleting its min and comparing it to the min of the list. It just keeps adding elements to the datastructures each iteration to hold their size. 

# Javadoc space:

Unfortunately i dont quite get what u mean by the javadoc spacing. When i autoformat it with eclipse it just goes back to how it was. Does the extra space really matter?
If it does, would you please remove it?

File: GAD 8 - Binomial Heap/tests/binomilia/BinomialHeapTester.java
Comment (N0W0RK) [reply GAD 8 - Binomial Heap/tests/binomilia/BinomialHeapTester.java:82]: I see overlooked something.

**Review by N0W0RK — APPROVED** [review]

**Comment by N0W0RK** [comment]
Basically, your indents between the stars and the beginning of your comment starting at line #19 use spaces. If you want to learn the reasoning behind this, have a look [here](https://checkstyle.sourceforge.io/config_whitespace.html#FileTabCharacter).

Sadly I can't edit the file before it is merged onto the repo.

**Review by N0W0RK — CHANGES_REQUESTED** [review]
The package also has to be changed to `tests.binomilia`

**Review by N0W0RK — DISMISSED** [review]

**Comment by CrsiX** [comment]
I just fixed the small point you mentioned, @N0W0RK. So far, I would say we can just merge the PR (I know that there are some remaining issues with tabs and spaces, but this could be addressed in #49, too).

Thanks, @maharajamihir for the work you did :)

**Comment by CrsiX** [comment]
Okay. So, the remaining issues have been fixed now, finally :)
I will merge and close this PR now.

**Review by CrsiX — APPROVED** [review]

**Pull request merged**
